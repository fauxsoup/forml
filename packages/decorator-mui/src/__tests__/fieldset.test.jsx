import FieldSet from '../fieldset';
import Context from '@forml/context';
import React from 'react';
import { render } from '@testing-library/react';
import * as decorator from '../';

describe('renders', function () {
    let form;
    let title = 'title';
    let description = 'description';

    beforeEach(function () {
        form = { type: 'fieldset', items: [{ key: [] }] };
    });

    describe('with form options', function () {
        let fields = {
            layout: ['horizontal', 'vertical'],
            disableGutters: [true, false],
            disablePadding: [true, false],
            disableMargin: [true, false],
            fullWidth: [true, false],
            Component: ['div', 'fieldset'],
            elevation: [0, 1, 2],
            icon: ['person', 'favorite'],
        };

        Object.keys(fields).forEach(function (field) {
            fields[field].forEach(function (value) {
                describe(`${field}`, function () {
                    test(`${value}`, function () {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator }}>
                                <FieldSet
                                    form={form}
                                    title={title}
                                    description={description}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });

    test('with title and description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} title={title} description={description} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with title and no description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} title={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with description and no title', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} description={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with no title or description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <FieldSet form={form} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
