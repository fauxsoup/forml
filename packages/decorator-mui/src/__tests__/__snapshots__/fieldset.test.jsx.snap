// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with description and no title 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options Component div 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options Component fieldset 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options disableGutters false 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options disableGutters true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options disableMargin false 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options disableMargin true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options disablePadding false 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options disablePadding true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-16twzsx"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options elevation 0 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-c5pdtz-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options elevation 1 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options elevation 2 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-z4i9t5-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options fullWidth false 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options fullWidth true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options icon favorite 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options icon person 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options layout horizontal 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-10au80u"
        layout="horizontal"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with form options layout vertical 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
  >
    <div
      class="MuiBox-root css-rwzwky"
      layout="vertical"
    />
  </ul>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1lskhvk-MuiPaper-root"
  >
    <ul
      class="MuiList-root MuiList-dense css-183iplg-MuiList-root"
    >
      <div
        class="MuiBox-root css-rwzwky"
        layout="vertical"
      />
    </ul>
  </div>
</div>
`;
